server/mscalendar/availability.go

```diff
@@ -106,6 +112,37 @@ func (m *mscalendar) setUserStatuses(users store.UserIndex, schedules []*remote.
 	return utils.JSONBlock(schedules), nil
 }
 
+func (m *mscalendar) notifyUpcomingEvent(mattermostUserID string, items []remote.ScheduleItem) {
+	var timezone string
+	for _, scheduleItem := range items {
+		upcomingTime := time.Now().Add(upcomingEventNotificationTime)
+		start := scheduleItem.Start.Time()
+		diff := start.Sub(upcomingTime)
+
+		if (diff < upcomingEventNotificationWindow) && (diff > -upcomingEventNotificationWindow) {
+			var err error
```

Comment:
Any reason why we need to declare this here?

Reply:
It is one of the multiple workarounds to one weird issue with multiple variable declaration.
I need timezone to be declared outside of the for loop. But I need to give it value inside the `if timezone == ""`.
If I try to assign err inside the if like `timezone, err = ...` will complain about err not being declared. If I do the implicit declaration with `timezone, err := ...` it will also create a new variable timezone that will shadow my previous one.
Since I am going to use the variable err in more places, I thought it ok to declare it before the if, but it would make sense only inside the if.
Another approach would be doing something like `aux, err := ...` and then `timezone = aux`.